README.md

```diff
@@ -2,6 +2,7 @@
 [![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
 [![ci](https://github.com/Deltares/HYDROLIB-core/actions/workflows/ci.yml/badge.svg)](https://github.com/Deltares/HYDROLIB-core/actions/workflows/ci.yml)
 [![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=Deltares_HYDROLIB-core&metric=alert_status)](https://sonarcloud.io/dashboard?id=Deltares_HYDROLIB-core)
+[![codecov](https://img.shields.io/codecov/c/github/deltares/HYDROLIB-core.svg?style=flat-square)](https://app.codecov.io/gh/deltares/HYDROLIB-core?displayType=list)
```

Comment:
I also propose to add the codecov badge to the readme to provide users with this valuable insight